urdf/include/common_macros.urdf.xacro

```diff
@@ -45,6 +45,76 @@
+      <origin xyz="${wheel_base_dx} ${-caster_wheel_base_dy * locationright} ${caster_wheel_base_dz}" rpy="0 0 0" />
+      <parent link="base_link" />
+      <child link="${locationprefix}_caster_rotation_link" />
+      <axis xyz="0 0 1" />
+      <dynamics damping="0.01" friction="0.0"/>
+    </joint>
+    <link name="${locationprefix}_caster_rotation_link">
+      <inertial>
+        <origin xyz="0 0 ${-caster_base_size/2.0}" rpy="0 0 0" />
+        <mass value="${caster_base_mass}" />
+        <xacro:box_inertia  m="${caster_base_mass}" l="${caster_base_size}" w="${caster_base_size}" h="${caster_base_size}" />
+      </inertial>
+      <visual>
+        <origin xyz="0 0 ${-caster_base_size/2.0}" rpy="0 0 0" />
+        <geometry>
+          <!--mesh filename="package://mobile_robot_description/meshes/visual/caster_wheel_base.stl" /-->
+          <box size="${caster_base_size} ${caster_base_size} ${caster_base_size}" />
+        </geometry>
+        <xacro:insert_block name="material_silver" />
+      </visual>
+      <collision>
+        <origin xyz="0 0 ${-caster_base_size/2.0}" rpy="0 0 0" />
+        <geometry>
+          <!--mesh filename="package://mobile_robot_description/meshes/collision/caster_wheel_base.stl" /-->
+          <box size="${caster_base_size} ${caster_base_size} ${caster_base_size}" />
+        </geometry>
+      </collision>
+    </link>
+    <gazebo reference="${locationprefix}_caster_rotation_link">
+      <material>Gazebo/Grey</material>
+    </gazebo>
+    <!-- caster wheel -->
+    <joint name="${locationprefix}_caster_wheel_joint" type="continuous">
+      <origin xyz="${caster_wheel_dx} ${-caster_wheel_dy * locationright} ${caster_wheel_dz}" rpy="0 0 0" />
+      <parent link="${locationprefix}_caster_rotation_link" />
+      <child link="${locationprefix}_caster_wheel_link" />
+      <axis xyz="0 1 0" />
+    </joint>
+
+    <link name="${locationprefix}_caster_wheel_link">
+      <xacro:cylinder_inertial mass="${caster_wheel_mass}" radius="${caster_wheel_radius}" length="${caster_wheel_width}">
+        <origin xyz="0 0 0" rpy="${pi/2} 0 0" />
+      </xacro:cylinder_inertial>
+      <visual>
+        <origin xyz="0 0 0" rpy="${flipY*pi/2} 0 0" />
+        <geometry>
+          <!--mesh filename="package://mobile_robot_description/meshes/visual/wheel.stl" -->
+          <cylinder length="${caster_wheel_width}" radius="${caster_wheel_radius}" />
+        </geometry>
+        <xacro:insert_block name="material_dark_grey" />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="${flipY*pi/2} 0 0" />
+        <geometry>
+          <!--mesh filename="package://mobile_robot_description/meshes/visual/wheel.stl" -->
+          <cylinder length="${caster_wheel_width}" radius="${caster_wheel_radius}" />
+        </geometry>
+      </collision>
+    </link>
+    <gazebo reference="${locationprefix}_caster_wheel_link">
+        <material>Gazebo/DarkGrey</material>
+    </gazebo>
```

Comment:
Should we put the gazebo tags into a new include file?

Reply:
I don't think so because these tags need to be connected to a specific link name. We can leave like this.